### PR TITLE
docs: add links to TypeScript

### PR DESCRIPTION
**Pull request by AlexElin**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/forms.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ This tutorial teaches you how to do the following:
 
 Before going further into template-driven forms, you should have a basic understanding of the following.
 
-* TypeScript and HTML5 programming.
+* [TypeScript](https://www.typescriptlang.org/docs/home.html "The TypeScript language") and HTML5 programming.
```
Comment (gkalpak) [inline aio/content/guide/forms.md:45]: The URL seems to redirect to just `/docs`:

```suggestion
* [TypeScript](https://www.typescriptlang.org/docs "The TypeScript language") and HTML5 programming.
```

I would also consider linking to the homepage instead:

```suggestion
* [TypeScript](https://www.typescriptlang.org/ "The TypeScript language") and HTML5 programming.
```

(Same for the other link in `reactive-forms.md`.)

Comment (gkalpak) [reply aio/content/guide/forms.md:45]: I just realized there are a couple of places that link to https://www.typescriptlang.org/docs/home.html in guides. I would suggest changing them too.

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx, @AlexElin! I've made a suggestion to change to URL, but otherwise I believe this addition can be quite useful for newcomers :+1:

File: aio/content/guide/forms.md (hunk 1 above)
Comment (AlexElin) [reply aio/content/guide/forms.md]: I've changed to the homepage.
Except `index.md`, in which I've changed to `/docs`

File: aio/content/start/index.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ To get the most out of this tutorial you should already have a basic understandi
 
 * [HTML](https://developer.mozilla.org/en-US/docs/Learn/HTML "Learning HTML: Guides and tutorials")
 * [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript "JavaScript")
-* [TypeScript](https://www.typescriptlang.org/docs/home.html "TypeScript documentation")
+* [TypeScript](https://www.typescriptlang.org/docs "TypeScript documentation")
```
Comment (gkalpak) [inline aio/content/start/index.md:16]: OOC, why not change this one too?

**Review by gkalpak — APPROVED** [review]
LGTM (with one question) 🚀 

Reviewed-for: global-docs-approvers

File: aio/content/start/index.md (hunk 2 above)
Comment (AlexElin) [reply aio/content/start/index.md:16]: Because the link text says "TypeScript documentation" I've thought the link should navigate directly to the docs

Comment (gkalpak) [reply aio/content/start/index.md:16]: I think it makes more sense to change the `tooltip` text and link to the homepage instead.

Comment (gkalpak) [reply aio/content/start/index.md:16]: I think it makes more sense to change the `tooltip` text and link to the homepage instead.

WDYT?

**Review by AndrewKushnir — APPROVED** [review]

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**